service/collector/tiller_reachable.go

```diff
@@ -19,16 +19,28 @@ var (
 )
 
 func (c *Collector) collectTillerReachable(ctx context.Context, ch chan<- prometheus.Metric) {
-	c.logger.LogCtx(ctx, "level", "debug", "message", "collecting Tiller reachability")
-
-	err := c.helmClient.PingTiller(ctx)
 	var value float64
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", "collecting Tiller reachability")
+	chartConfigs, err := c.getChartConfigs()
 	if err != nil {
-		c.logger.Log("level", "error", "message", "could not ping Tiller", "stack", fmt.Sprintf("%#v", err))
+		c.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("could not get ChartConfigs"), "stack", fmt.Sprintf("%#v", err))
```

Comment:
Can we just return here? The conditions below give me a head ache. Can we make this simpler and better understandable? When we can return here in case there is an error, we only have to check for the number of chart configs below.

Reply:
Returning here would simplify things but also it would trigger `TillerUnreachable` which does not feels right.
A solution can be to set the metric value to 1 also in case we are not able to get ChartConfigs.

So we would actually only ping tiller when we have existing ChartConfigs, and skip otherwise.

Reply:
We should only set 1 in case tiller is reachable as this is what the metric is about.

Reply:
Ok, but in the other hand setting 0 in case we are not sure, is something I would like to avoid as this mean triggering `TillerUnreachable` alert with a potentially false positive.

My original thinking here was: chart-operator is responsible for installing tiller whenever it needs it (when there is ChartConfigs to reconcile). So chart-operator should only check for tiller ping in case it needs it. In case chart-operator is not able to determine if there's any ChartConfigs let's fallback to tiller ping as throwing either 0 or 1 would seems wrong.

Reply:
What is the case of fetching chart configs causing errors? How likely is it for such errors to occur? I would presume these events are the rather least likely ones to happen. So returning an error in such cases in a collector implementation makes sense. The error rate alert for the operator should help us getting after it once errors keep happening over a longer period of time. 

What is actually the decision making rule that Tiller is installed or not? Are the chart configs the resources we have to check here or is there something else? 

My initial point here was also to just make the implementation a bit simpler and not have a lot of wild, hardly to understand conditions that deviate from the norm.

Reply:
I did added the return in case we are not able to get chart configs, the outcome is the same as the previous condition but more readable and as you pointed out that's also very unlikely to happen.

> What is actually the decision making rule that Tiller is installed or not? Are the chart configs the resources we have to check here or is there something else?

Yes, we only need to check for chart configs.

PTAL